src/poetry/core/packages/package.py

```diff
@@ -506,6 +506,7 @@ def to_dependency(self) -> Dependency:
             dep = URLDependency(
                 self._name,
                 cast(str, self._source_url),
+                directory=self.source_subdirectory,
```

Comment:
That seems to be untested. Can you add a test to `test_package.py` please?

Reply:
Added a check in `test_to_dependency_for_url()`